### PR TITLE
remove multi-emit - internal restriction

### PR DESCRIPTION
**Pull request by KevinRansom**
The multi-emit pr: https://github.com/dotnet/fsharp/pull/12722

Introduced a limitation of 30 submissions visibility for any newly introduced internal value.

The reason for the limitation was that submission assemblies include 30 internals visible for assemblies with the name FSI_ASSEMBLY_nnnn ...

This PR removes the limitation by removing the naming pattern for assemblies and using assembly versioning for each assembly submission.

In order to make this work, assemblyresolvecore in fsi.fs was updated to do assembly fullname resolution when resolving dynamic assemblies.

It also removes the warning and adds a test.

I also did some more minor rework around signing to allow keypairs and keyfiles to be specified as raw byte arrays to be specified with bytearrays rather than merely from files.

### CODE REVIEW

**Review by KevinRansom — COMMENTED** [review]
oops

File: src/Compiler/Interactive/fsi.fs [hunk 1]
```diff
@@ -1406,44 +1405,29 @@ type internal FsiDynamicCompiler(
                  Some { man with  CustomAttrsStored = storeILCustomAttrs (mkILCustomAttrs codegenResults.ilAssemAttrs) }) }
 
     /// Generate one assembly using multi-assembly emit
-    let EmitInMemoryAssembly (tcConfig: TcConfig, emEnv: ILMultiInMemoryAssemblyEmitEnv, ilxMainModule: ILModuleDef, m) =
+    let EmitInMemoryAssembly (tcConfig: TcConfig, emEnv: ILMultiInMemoryAssemblyEmitEnv, ilxMainModule: ILModuleDef) =
         
         // The name of the assembly is "FSI-ASSEMBLY1" etc
```
Comment (T-Gro) [inline src/Compiler/Interactive/fsi.fs:1410]: The comment is no longer correct, right?

In the new style, the naming is agnostics of the number (therefore only 1 IVT attribute needed), and the number just makes it to the version.

File: src/Compiler/Interactive/fsi.fs [hunk 2]
```diff
@@ -1479,6 +1463,7 @@ type internal FsiDynamicCompiler(
 
         let assemblyBytes, pdbBytes = WriteILBinaryInMemory (opts, ilxMainModule, normalizeAssemblyRefs)
 
+        File.WriteAllBytes($"""c:\temp\FSI-ASSEMBLY-{dynamicAssemblyId}""", assemblyBytes)
```
Comment (T-Gro) [inline src/Compiler/Interactive/fsi.fs:1466]:
```suggestion
     
```

I assume this was just test code...

**Review by KevinRansom — COMMENTED** [review]
oops

File: src/Compiler/Interactive/fsi.fs [hunk 3]
```diff
@@ -1406,44 +1405,29 @@ type internal FsiDynamicCompiler(
...
         
         // The name of the assembly is "FSI-ASSEMBLY1" etc
         dynamicAssemblyId <- dynamicAssemblyId + 1
-
-        let multiAssemblyName = ilxMainModule.ManifestOfAssembly.Name + string dynamicAssemblyId
+ 
+        let multiAssemblyName = ilxMainModule.ManifestOfAssembly.Name
 
         // Adjust the assembly name of this fragment, and add InternalsVisibleTo attributes to 
         // allow internals access by multiple future assemblies
```
Comment (T-Gro) [inline src/Compiler/Interactive/fsi.fs:1416]:
```suggestion
        // allow internals access by all future assemblies with the same name (and only differing in version)
```

File: src/Compiler/Interactive/fsi.fs (hunk 1 above)
Comment (T-Gro) [inline src/Compiler/Interactive/fsi.fs:1410]:
```suggestion
        // The name of the assembly is "FSI-ASSEMBLY" for all submissions. This number is used for the Version 
```

File: src/Compiler/Interactive/fsi.fs (hunk 2 above)
Comment (KevinRansom) [reply src/Compiler/Interactive/fsi.fs]: done

File: src/Compiler/Interactive/fsi.fs (hunk 1 above)
Comment (KevinRansom) [reply src/Compiler/Interactive/fsi.fs]: fixed, thanks

**Review by T-Gro — APPROVED** [review]

**Review by vzarytovskii — APPROVED** [review]

**Pull request merged**